### DISCUSSION TITLE
Copy DataGrid Row to Clipboard

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#5903] [Q&A] [answered]
I have a Datagrid with Context Menu to allow user to copy DataGrid Row and Cell values to Clipboard. Currently I am using the `DataGridCellPointerPressedEventArgs` event to extract the row and cell information. I get to copy the cell value to clipboard using the below code. However, I am not sure how to copy single/ multiple Row with/without Headers to clipboard. I tried including `ClipboardCopyMode="IncludeHeader"` in my xaml but that isn't copying anything. Is there any way to do that?

```
private void CopyRow(object sender, DataGridCellPointerPressedEventArgs e)
        {
            var dataGrid = sender as DataGrid;
            var currentRow = e.Row;
            var currentCell = e.Cell.Content as TextBlock;
            Application.Current.Clipboard.SetTextAsync(currentCell.Text);
        }
```

**Comment by FoggyFinder** [comment, marked as answer]
You could access underlying VM item that is bound to a row and use it instead of getting data from the Content. Alternatively you could pass `SelectedItems` via converter (as well as current `ClipboardCopyMode`) to your VM and set text from there.

It gives you additional abilities to format text easily etc.

**Reply by ShrutiJaiswal1494** [reply]
@FoggyFinder  Thanks for the reply. Apart from the row I also need the selected cell value. Is there a way to get  the selected cell value in View model? I tried binding to `CurrentColumn` Property of `DataGrid` but that was throwing error as Avalonia was not able to detect this property.

**Reply by FoggyFinder** [reply]
Nope, sadly there is no such thing like `SelectionUnit` yet. As soon as it will be implemented there will be easy way to do it. Right now I'm unsure how to do it properly since none of my applications required it.
